### ISSUE TITLE
Implement option not to shift the time scale at all when data is added with `setData`

### ISSUE THREAD

**Issue by nalepae** [tradingview/lightweight-charts#584]
**What we have today:**

1. Thanks to https://github.com/tradingview/lightweight-charts/pull/555, if new data is added to the left with `setData`, then the time scale doesn't shift.
2. If new data is added to the right with `setData` **and** if the last series bar is visible, then the time scale doesn't shift.
3. If new data is added to the right with `setData` **and** if the last series bar is **not** visible, then the time scale **does** shift.

**What is wished:**

I perfectly understand the use case of the point 3.
But sometimes, it is needed that even if the last series bar is **not** visible, the time scale would **not** shift when data is added to the right with `setData`.

An idea would be to add a boolean option like `neverShiftTimeScaleWhenDataIsAdded` defaulted to `false`.

In the following example, my program adds automatically new data to the **left** when it detects a "hole" between the left border of the graph and the first candle. We can see the new data is added whereas the time scale remains the same.

![Kapture 2020-09-29 at 18 10 29](https://user-images.githubusercontent.com/4943830/94584557-161c7d00-027f-11eb-815e-4ce0b6afca56.gif)

In the following example, my program adds automatically new data to the **right** when it detects a "hole" between the last candle and the right border of the graph. If there is a "hole", then just after the `setData`, the timescale is automatically moved to the right, so there is a new "hole", so new data is added, so the timescale is automatically moved to the right, so there is a new "hole" ...

![Kapture 2020-09-29 at 18 30 39](https://user-images.githubusercontent.com/4943830/94586985-3f8ad800-0282-11eb-9e19-1c8144a46dba.gif)

For information, the desired behaviour is reached when, in file `model/chart-model.ts`, in function `updateTimeScale`, I replace:

```typescript
if (isSeriesPointsAddedToRight && !isLastSeriesBarVisible) {
	const compensationShift = newBaseIndex - currentBaseIndex;
	this._timeScale.setRightOffset(
		this._timeScale.rightOffset() - compensationShift
	);
}
```

by
```typescript
if (isSeriesPointsAddedToRight) {
	const compensationShift = newBaseIndex - currentBaseIndex;
	this._timeScale.setRightOffset(
		this._timeScale.rightOffset() - compensationShift
	);
}
```

(I removed the condition `&& !isLastSeriesBarVisible`)

**Comment by nalepae** [comment]
I am not sure about `need more feedback` means, so I added some additional information in my ticket.
Please tell me if something is not clear!

**Comment by timocov** [comment]
> so I added some additional information in my ticket.

Thanks a lot! `need more feedback` means that we need to wait some more feedback from users who needs this feature as well to understand priorities. Added examples clears things for me now and now I understand use-case.

> An idea would be to add a boolean option like neverShiftTimeScaleWhenDataIsAdded defaulted to false.

I think that we need to re-phase this option somehow, but the idea will be the same. I think that we have 2 ways if you'd like to make a PR for this feature:

1. You can implement this feature with the name you've referenced, but quite probably we'll ask to rename it somehow (I don't think that it will be big deal to rename it though)
1. I can ask other maintainers to look at this and suggest any better name if they have one, and then do 1.

2nd means that I need to label this issue as "need proposal" basically, but this change doesn't look like something big so I don't think that renaming the option will be a problem.

What do you think?

**Comment by paulocentr** [comment]
@timocov  Hello how are you?
I'm sorry posting this here, but i have no idea on how to contact ppl from Trading view,
I'm applyed for charting library 
https://github.com/tradingview/charting_library
and got an invitation, but the link expired before i could even accept, do you mind resending the invitation?
or at least telling where i can submit this request (took almost a month to get this invitation and no email has come)
this is my github (paulocentr)
here a print from the invitation before i can even accept
https://postimg.cc/LJz6SbLN

**Comment by nalepae** [comment]
🙏🙏🙏 !

**Comment by timocov** [comment]
The fix is just published in [3.3.0 version](https://github.com/tradingview/lightweight-charts/releases/tag/3.3.0).
